core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -332,6 +332,34 @@ package object config {
     ConfigBuilder("spark.dynamicAllocation.sustainedSchedulerBacklogTimeout")
       .fallbackConf(DYN_ALLOCATION_SCHEDULER_BACKLOG_TIMEOUT)
 
+  private[streaming] val STREAMING_DYN_ALLOCATION_ENABLED =
```

Comment:
It's good to refactor the streaming configs into config objects like this. However I think these should do into a new `.../streaming/config/package.scala` file. Right now I'm not sure how it would interpret `private[streaming]` from here.

Really, this is most of the change here. It's OK to do in one place, but I'd open a new JIRA as a subtask of https://issues.apache.org/jira/browse/SPARK-26442 for streaming (can't believe we missed that!) and also add that new JIRA to the title.

There may be more. You could also fix additional `spark.streaming.` configs separately.

Reply:
@srowen OK, thanks for your suggestion, I will follow the new subtask after finish this PR.

Reply:
Yes you can open a new subtask, at least, and continue with another PR if you're so inclined.